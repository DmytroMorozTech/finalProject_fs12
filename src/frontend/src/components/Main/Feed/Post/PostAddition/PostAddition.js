import { MenuItem } from '@material-ui/core'
import React from 'react'
import styles from './styles'
import { toggleBookmarkAction } from '../../../../../redux/Post/postActions'
import { useDispatch } from 'react-redux'
import BookmarkBorderIcon from '@material-ui/icons/BookmarkBorder'
import BookmarkIcon from '@material-ui/icons/Bookmark'
import Typography from '@material-ui/core/Typography'

function PostAddition (props) {
  const classes = styles()

  const { postId, isBookmarkedByActiveUser, feedType } = props

  const dispatch = useDispatch()
  const handleBookmarkPost = () => {
    dispatch(toggleBookmarkAction(postId, feedType))
  }

  return (
    <div>
      <MenuItem className={classes.menuItem}>
        <div
          id='item'
          className={classes.item}
          onClick={handleBookmarkPost}>
          <div className={classes.icons}>
            {isBookmarkedByActiveUser ? <BookmarkIcon fontSize="inherit"/> : <BookmarkBorderIcon fontSize="inherit"/>}
          </div>
          <div className={classes.titles}>
            <Typography variant="h5">
              {isBookmarkedByActiveUser ? 'Unsave' : 'Save'}
            </Typography>
            <Typography variant="h6">
              {isBookmarkedByActiveUser ? 'Unsave from your saved list' : 'Save for later'}
            </Typography>
          </div>
        </div>
      </MenuItem>
    </div>
  )
}

export default PostAddition
