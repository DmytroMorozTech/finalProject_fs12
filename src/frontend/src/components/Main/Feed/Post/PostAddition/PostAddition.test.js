import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import PostAddition from './PostAddition'

jest.mock('./styles', () => () => ({}))

const setUp = (props) => shallow(<PostAddition {...props} />)

describe('should render PostAddition component', () => {
  let spyOnUseDispatch
  let mockDispatch

  let feedType
  let postId
  let isBookmarkedByActiveUser

  let component
  beforeEach(() => {
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    postId = 2
    feedType = 'posts'
    isBookmarkedByActiveUser = false

    component = setUp({postId, isBookmarkedByActiveUser, feedType})
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
  it('to render BookmarkBorderIcon', () => {
    expect.stringMatching('BookmarkBorderIcon')
  })
})

describe('should render PostAddition component', () => {
  let spyOnUseDispatch
  let mockDispatch

  let feedType
  let postId
  let isBookmarkedByActiveUser

  let component
  beforeEach(() => {
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    postId = 2
    feedType = 'posts'
    isBookmarkedByActiveUser = true

    component = setUp({postId, isBookmarkedByActiveUser, feedType})
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to render BookmarkIcon', () => {
    expect.stringMatching('BookmarkIcon')
  })
  it('should work toggleBookmark dispatch', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('div[id=\'item\']').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})
