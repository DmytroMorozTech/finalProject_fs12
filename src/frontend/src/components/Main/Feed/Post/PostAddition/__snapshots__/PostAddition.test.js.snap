// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PostAddition component to match snapshot 1`] = `
<div>
  <WithStyles(ForwardRef(MenuItem))>
    <div
      id="item"
      onClick={[Function]}
    >
      <div>
        <Memo(ForwardRef(BookmarkBorderIcon))
          fontSize="inherit"
        />
      </div>
      <div>
        <WithStyles(ForwardRef(Typography))
          variant="h5"
        >
          Save
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="h6"
        >
          Save for later
        </WithStyles(ForwardRef(Typography))>
      </div>
    </div>
  </WithStyles(ForwardRef(MenuItem))>
</div>
`;
